site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+
+For the already-existing `parents` stanza, the status is effectively
+namespaced by two things: the `parentRef` and the `controllerName`.
+An example is included below of error handling when routes are attached to multiple Gateways with different GatewayClasses (and thus `controllerNames`).
+
+In the case that an implementation sees a status that does *not* match its `controllerName` field, it *must* put that status back unaltered.
+To put this another way, an implementation *must* only modify `parents` entries that match its `controllerName`. (This is implied by the base spec, but not well documented at the moment).
+
+### Status additions
+
+The basic design:
+- we'll add a `grandparents` stanza inside the parents stanza in status, which will contain details about the route being a parent, successfully or not.
+- the `grandparents` stanza will only be populated if a rule has a non-nil `allowedRoutes` section.
+- once the `grandparents` stanza is populated, Conditions are used to describe if the configuration still generates a valid Ephemeral Route configuration.
+
+Here's what the proposal looks like for an attached child route:
+
+```yaml
+status:
+  # Existing stanza is the same
+  parents:
+  - parentref:
+      name: somegateway
+      namespace: somenamespace
+    controllerName: controller.io/somecontroller
+    conditions:
+    - <conditions here>
+    grandparents:
+    - conditions:
+      - <conditions here>
+```
+We add a new `grandparents` stanza inside the the top-level `parents` status. This stanza will be `+optional` and `omitempty`, with the following rules:
+- `grandparents` will be nil in all non-parent HTTPRoutes.
+- Once a HTTPRoute is a parent, then this stanza must also be non-nil.
+
+We need to add this inside the parents stanza because it's possible that a parent HTTPRoute may be attached to more than one parent (probably Gateways) of its own, and there may be different status between different Gateways. For example, if a parent HTTPRoute is attached to two Gateways, one of which has a wildcard hostname, whether or not the child HTTPRoute attempting to attach to the parent HTTPRoute will be accepted is dependent on the hostname setting on the Gateway Listener. An example demonstrating this is included below.
+
+The struct itself has the same `parentRef` and `conditions` fields as the `parents` struct.
+
+The `parentRef` in the `grandparents` struct must be the same as the one on the *parent* Route's status. See the example for more clarity here.
+
+For Conditions:
+- when the controller first sees the parent Route, it must add the following Condition Types to the *parent* Route:
+  - `Attached`: one or more Routes have successfully attached to this parent. False if zero Routes have attached successfully. Note that a new Reason has also been added, `PartiallyAttached` which indicates that *some* routes have tried to attach and failed.
```

Comment:
This is the only mention of PartiallyAttached, is it meant to be included in more detail somewhere?

Reply:
It's a Reason, not a Condition, so I think we only ended up mentioning it here. I agree that this section could use more fleshing out though.